citesphere/src/main/java/edu/asu/diging/citesphere/core/service/giles/impl/GilesUploadCheckerImpl.java

```diff
@@ -219,4 +182,90 @@ public ICitation getCurrentCitation(ICitation citation, IUser user) {
         }
         return null;
     }
+    
+    private Set<IGilesUpload> getFileStatus(IGilesUpload upload, IUser user) {
+        Set<IGilesUpload> checkedUploads = new HashSet<>();
+        if (upload.getUploadingUser() == null
+                || Arrays.asList(GilesStatus.COMPLETE, GilesStatus.FAILED)
+                        .contains(upload.getDocumentStatus())) {
+            // in case something went wrong with the user
+            // or the upload has been processed
+            return null;
+        }
+        
+        if(user == null) {
```

Comment:
in what case would this be true? and should Citesphere then simply use the uploaded users token (instead of the token of the user that is checking the status)?

Reply:
in other words, if this is a valid scenario there needs to be some explanation here.